Client/Frontend/Widgets/SnackButton.swift

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/**
+ * A specialized version of UIButton for use in SnackBars. These are displayed evenly
+ * spaced in the bottom of the bar. The main convenience of these is that you can pass
+ * in a callback in the constructor (although these also style themselves appropriately).
+ */
+typealias SnackBarCallback = (_ bar: SnackBar) -> Void
+class SnackButton: UIButton {
+    let callback: SnackBarCallback?
+    var bar: SnackBar!
+
+    private struct SnackBarUX {
+        static let BorderWidth: CGFloat = 0.5
+    }
+
+    override open var isHighlighted: Bool {
+        didSet {
+            self.backgroundColor = isHighlighted ? UIColor.legacyTheme.snackbar.highlight : .clear
+        }
+    }
+
+    init(title: String, accessibilityIdentifier: String, bold: Bool = false, callback: @escaping SnackBarCallback) {
+        self.callback = callback
+
+        super.init(frame: .zero)
+
+        if bold {
+            titleLabel?.font = DynamicFontHelper.defaultHelper.DefaultStandardFontBold
+        } else {
+            titleLabel?.font = DynamicFontHelper.defaultHelper.DefaultStandardFont
+        }
+        titleLabel?.adjustsFontForContentSizeCategory = false
```

Comment:
We should use Dynamic Text and with that we should adjusts fonts when the content size category changes. This will probably require testing of all the place we use the button.